packages/esm-service-queues-app/src/pages/queue-table-by-status-page.component.tsx

```diff
@@ -0,0 +1,21 @@
+import { InlineNotification } from '@carbon/react';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import { useQueue } from '../hooks/useQueue';
+import { QueueTableByStatus } from '../queue-table/queue-table-by-status.component';
+
+export const QueueTableByStatusPage: React.FC = () => {
+  const { queueUuid, statusUuid } = useParams();
+  const { queue, isLoading } = useQueue(queueUuid);
+  const status = queue?.allowedStatuses.find((s) => s.uuid == statusUuid);
+  const { t } = useTranslation();
+
+  if (isLoading) {
+    return <>{t('loading', 'Loading...')}</>;
+  } else if (!queue) {
+    return <InlineNotification kind="error" title={t('invalidQueue', 'Invalid Queue')} />;
+  } else {
+    return <QueueTableByStatus queue={queue} status={status} />;
+  }
```

Comment:
I don't think the `loading...` thing actually confirms to Carbon design, but it is also in lots of places. They have `<SkeletonPlaceholder/>` for that (It's the blurred out tables/texts on some pages while they are loading). I just tried to play around with it but have CSS trouble with it to make it look nice.

Reply:
100% agree that it would be import to distinguish between top-level pages vs components.  Not sure if there's a best practice around the best way to do this, but it definitely would have been helpful when looking through the appointments code.  

I like the idea of the "*Page" suffix on the end of page components (which is the case here, but not in many of the components, at least in the Appointments ESM), that might be preferrable to using a pages directory because it would still allow you to group components together functionally. ie the page and it's children components could all be in the same directory (for child components that it wouldn't make to live in some common components directory)

Yeah, seemed like there should be some carbon design styling around this (sounds like there is and we just have problems using it).

Reply:
So, we already have a concept in core called a "page" (in most apps, it's just the thing called `root.component.tsx`). I wouldn't want to re-use that name, as it will add to the confusion.

Reply:
@ibacher in the case of the queues app, the route component contains this:

```
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/queue-table-by-status/:queueUuid" element={<QueueTableByStatusPage />} />
            <Route path="/queue-table-by-status/:queueUuid/:statusUuid" element={<QueueTableByStatusPage />} />
            <Route path="/screen" element={<QueueScreen />} />
            <Route path="/appointments-list/:value/" element={<AppointmentsTable />} />
            <Route path="/queue-list/:service/:serviceUuid/:locationUuid" element={<ServicesTable />} />
          </Routes>
```

These are the things I'm calling `pages`.  Isn't that how the end user sees them?  I mean, we are in a single-page app, so they aren't really pages, but if they show new content on the screen and they are accessible with a specific URL, then they seem like things we'd want to identify as "pages", and distinct from the various components that one might add to them.  What would you suggest?

Reply:
So, it's true that they aren't true pages. I'm just being picky about terminology.

In `routes.json` for example, there's an element called `extensions` and there's also an element called `pages`. These `pages` are relatively rare. For example, everything in the `/home` route is rendered in a single "page" defined [here](https://github.com/openmrs/openmrs-esm-home/blob/f4e0609219e40f9fba26aa23a91690905ea44a2c/packages/esm-home-app/src/routes.json#L6); similarly, everything in patient chart uses [a single page](https://github.com/openmrs/openmrs-esm-patient-chart/blob/4a9a90edf7500ba2d1e0ccee7d218425114bcec3/packages/esm-patient-chart-app/src/routes.json#L239). So, while from the user's perspective the elements render into routes may also appear like "pages", I don't want to muddle terminology too much. (The "pages" terminology pre-dates `routes.json`; in the migration to v5, we just refactored existing pages into JSON definitions from JS definitions).

I think the naming we have for what you're talking about (a component that takes over the entire "page") is a "dashboard". (Both the chart and home page internally call these things "dashboards"). It's a little confusing, because technically, the whole service queues app is a "dashboard" and the routes within it are just the dashboard contents.

Reply:
@mseaton  you can fill me in on how everything works tomorrow... :)